app/controllers/cars_controller.rb

```diff
@@ -1,7 +1,29 @@
 class CarsController < ApplicationController
   skip_before_action :authenticate_user!, only: %i[index]
-
+  before_action :set_user, only: %i[new create]
   def index
     @cars = Car.all
   end
+
+  def new
+    # @user = User.find(params[:user_id])
+    @car = Car.new
+  end
+
+  def create
+    @car = Car.new(car_params)
+    @car.user = @user
+    @car.save
```

Comment:
Can you please update to check if its valid. Have a look at the following to see an example: https://github.com/florian-lefebvre/rails-watch-list-with-images/blob/master/app/controllers/bookmarks_controller.rb#L12-L16